Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1877,6 +1877,9 @@ void perform_url_and_history_update_steps(DOM::Document& document, AK::URL new_u
         // 1. Finalize a same-document navigation given traversable, navigable, newEntry, and entryToReplace.
         finalize_a_same_document_navigation(*traversable, *navigable, new_entry, entry_to_replace);
     });
+
+    // NOTE: Not in the spec but we need to make sure that history traversal queue is done to take care of race conditions
+    traversable->process_session_history_traversal_queue();
```

Comment:
Ah, this should be attached with a FIXME. the proper thing to do is add `synchronous history update steps` as in this WIP commit. https://github.com/SerenityOS/serenity/pull/21254/commits/a75650993df2532f5c3f05856fdde28b062b0ae1

Reply:
thank you! changed the comment to FIXME with a reference to the commit